client/components/ManageTestQueue/ManageAtVersions.jsx

```diff
@@ -0,0 +1,451 @@
+
+    const [selectedAtId, setSelectedAtId] = useState('1');
+    const [selectedAtVersions, setSelectedAtVersions] = useState([]);
+    const [selectedAtVersionId, setSelectedAtVersionId] = useState('');
+
+    const [addAtVersion] = useMutation(ADD_AT_VERSION_MUTATION);
+    const [editAtVersion] = useMutation(EDIT_AT_VERSION_MUTATION);
+    const [deleteAtVersion] = useMutation(DELETE_AT_VERSION_MUTATION);
+
+    // Update modal state values
+    const [showUpdateVersionModal, setShowUpdateVersionModal] = useState(false);
+    const [updateVersionModalTitle, setUpdateVersionModalTitle] = useState('');
+    const [updateVersionModalType, setUpdateVersionModalType] = useState('add');
+    const [updateVersionModalVersionText, setUpdateVersionModalVersionText] =
+        useState('');
+    const [
+        updateVersionModalModalDateText,
+        setUpdateVersionModalModalDateText
+    ] = useState('');
+
+    // Feedback modal state values
+    const [showFeedbackModal, setShowFeedbackModal] = useState(false);
+    const [feedbackModalTitle, setFeedbackModalTitle] = useState('');
+    const [feedbackModalContent, setFeedbackModalContent] = useState(<></>);
+
+    useEffect(() => {
+        if (ats.length) {
+            if (!loadedAts.current) setSelectedAtId(ats[0].id);
+
+            // Required during refetch logic around managing AT Versions
+            if (!loadedAts.current) setSelectedAtVersions(ats[0].atVersions);
+            else {
+                setSelectedAtVersions(
+                    ats.find(item => item.id === selectedAtId).atVersions
+                );
+            }
+
+            if (!loadedAts.current)
+                setSelectedAtVersionId(ats[0]?.atVersions[0]?.id);
+            loadedAts.current = true;
+        }
+    }, [ats]);
+
+    const getAtVersionFromId = id =>
+        selectedAtVersions.find(item => id === item.id);
+
+    const showThemedMessage = ({
+        title,
+        content,
+        theme,
+        actions = null,
+        showCloseAction = false
+    }) => {
+        setThemedModalTitle(title);
+        setThemedModalContent(content);
+        setThemedModalType(theme);
+        setThemedModalActions(actions);
+        setThemedModalShowCloseAction(showCloseAction);
+        setShowThemedModal(true);
+    };
+
+    const showFeedbackMessage = (title, content) => {
+        setFeedbackModalTitle(title);
+        setFeedbackModalContent(content);
+        setShowFeedbackModal(true);
+    };
+
+    const onAtChange = e => {
+        const { value } = e.target;
+        if (selectedAtId !== value) {
+            setSelectedAtId(value);
+            const at = ats.find(item => item.id === value);
+            setSelectedAtVersions(at.atVersions);
+            setSelectedAtVersionId(at.atVersions[0].id);
+        }
+    };
+
+    const onAtVersionChange = e => {
+        const { value } = e.target;
+        setSelectedAtVersionId(value);
+    };
+
+    const onOpenAtVersionModalClick = type => {
```

Comment:
Curious why this pattern is used. Shorter independent functions for each one of these `type`s could improve readability

Reply:
Agreed, also a copy over but no issue with addressing before landing.